### ISSUE TITLE
IMAP OAuth Authentication (XOAUTH2)

### ISSUE THREAD

**Issue by freescout-helpdesk** [zendframework/zend-mail#252]
Hi,

Is it possible to add XOAUTH2-authentication to the Zend Mail IMAP? See https://github.com/google/gmail-oauth2-tools/issues/18

The [solution](https://github.com/google/gmail-oauth2-tools/blob/master/php/oauth2.php) suggested by Google is not perfect, as it first tries to connect via regular authentication and only after that performs OAuth-authenticaiton.

Here is the implementation of the IMAP OAuth for Zend Mail IMAP: https://github.com/ialert/imap-xoauth2. You just need to add it to your repo.

**Comment by weierophinney** [comment]
This repository has been closed and moved to laminas/laminas-mail; a new issue has been opened at https://github.com/laminas/laminas-mail/issues/1.
